pandas/core/indexes/category.py

```diff
@@ -390,6 +390,7 @@ def __contains__(self, key) -> bool:
         if is_scalar(key) and isna(key):
             return self.hasnans
 
+        hash(key)
```

Comment:
I assume this is to check that key is hashable. can you not type key as Hashable?

Reply:
that's not a run-time check